### PR TITLE
[dhctl] mirror: Pull only specific release, skip modules option

### PR DESCRIPTION
**Pull request by mvasl**
## Description

Added options that allow to download only specific versions of Deckhouse and modules, as well as skip loading modules when downloading the main Deckhouse distribution.

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: feature
summary: |
  Add options for dhctl mirror to pull only specific release and skip modules option.
impact_level: default
```

### CODE REVIEW

**Comment by mvasl** [comment]
Specific release without modules
Снимок экрана 2024-04-10 в 16 56 09
Снимок экрана 2024-04-10 в 16 55 33

File: docs/documentation/pages/DECKHOUSE-FAQ.md [hunk 1]
```diff
@@ -426,7 +426,8 @@ This feature is available in Enterprise Edition only.
 
 1. Pull Deckhouse images using the `dhctl mirror` command.
 
-   `dhctl mirror` pulls only the latest available patch versions of a minor Deckhouse release. For example, for Deckhouse 1.52, only one version, `1.52.10`, will be pulled, as it is sufficient to update Deckhouse from version 1.51.
+   By default `dhctl mirror` pulls only the latest available patch versions for every actual Deckhouse release and the current set of officially supplied modules.
```
Comment (kirkonru) [inline docs/documentation/pages/DECKHOUSE-FAQ.md:429]:
```suggestion
   By default, `dhctl mirror` pulls only the latest available patch versions for every actual Deckhouse release and the current set of officially supplied modules.
```

File: docs/documentation/pages/DECKHOUSE-FAQ.md [hunk 2]
```diff
@@ -447,6 +450,11 @@ This feature is available in Enterprise Edition only.
 
    > Note that `--min-version` parameter will be ignored if you specify version above current rock-solid channel.
 
+   It is also possible to download only one specific Deckhouse release by using the `--release=X.Y.Z` flag.
+   In this case, release channels will not be used and only one specific release will be downloaded.
```
Comment (kirkonru) [inline docs/documentation/pages/DECKHOUSE-FAQ.md:454]:
```suggestion
   You can also pull a single specific Deckhouse release by using the `--release=X.Y.Z` flag.
   In this case, no release channels will be used and only one specific release will be pulled.
```

File: docs/documentation/pages/DECKHOUSE-FAQ.md [hunk 3]
```diff
@@ -517,14 +525,21 @@ The steps below are necessary for manually loading images of modules connected f
 
 1. Pull module images from their source registry, defined as a `ModuleSource` resource, into a dedicated directory using the command `dhctl mirror-modules`.
 
-   `dhctl mirror-modules` pulls only versions of modules available in the module release channels at the time of copying.
+   `dhctl mirror-modules` pulls only versions of modules available in the module release channels at the time of copying if the `--modules-filter` flag was not set.
```
Comment (kirkonru) [inline docs/documentation/pages/DECKHOUSE-FAQ.md:528]:
```suggestion
   `dhctl mirror-modules` pulls only the module versions available in the module release channels at the time of copying unless the `--modules-filter` flag is set.
```

File: docs/documentation/pages/DECKHOUSE-FAQ.md [hunk 4]
```diff
@@ -517,14 +525,21 @@ The steps below are necessary for manually loading images of modules connected f
...
 
    The following command will pull module images from the source described in the `ModuleSource` resource located in the `$HOME/module_source.yml` file:
 
    ```shell
    dhctl mirror-modules -d /tmp/d8-modules -m /tmp/module_source.yml
    ```
 
+   To download only a set of specific modules of specific versions, use the `--modules-filter` flag, passing the set of required modules and their versions, separated by the `;` character.
```
Comment (kirkonru) [inline docs/documentation/pages/DECKHOUSE-FAQ.md:536]:
```suggestion
   To download only a specific set of modules of specific versions, use the `--modules-filter` flag followed by the list of required modules and their versions separated by the `;` character.
```

File: docs/documentation/pages/DECKHOUSE-FAQ.md [hunk 5]
```diff
@@ -517,14 +525,21 @@ The steps below are necessary for manually loading images of modules connected f
...
+   For example:
+
+   ```shell
+   dhctl mirror-modules -d /tmp/d8-modules -m /tmp/module_source.yml --modules-filter="deckhouse-admin:v1.0.0;deckhouse-admin:v1.3.3; sds-drbd:v0.0.1"
```
Comment (kirkonru) [inline docs/documentation/pages/DECKHOUSE-FAQ.md:540]: --modules-filter="deckhouse-admin:v1.0.0`;`deckhouse-admin:v1.3.3`; `sds-drbd:v0.0.1"

Is there a space needed between versions? In the example above, it is there in one case and missing in the other. Or does it make no difference? (This applies to the Russian version as well.)

File: docs/documentation/pages/DECKHOUSE-FAQ_RU.md [hunk 6]
```diff
@@ -518,14 +526,21 @@ Deckhouse поддерживает работу только с Bearer token-с
...
 
    ```shell
    dhctl mirror-modules -d /tmp/d8-modules -m /tmp/module_source.yml
    ```
 
+   Для загрузки только набора из определенных модулей конкретных версий, используйте параметр `--modules-filter`, передав набор необходимых модулей и их версий, разделенных символом `;`.
+   Например:
+
+   ```shell
+   dhctl mirror-modules -d /tmp/d8-modules -m /tmp/module_source.yml --modules-filter="deckhouse-admin:v1.0.0;deckhouse-admin:v1.3.3; sds-drbd:v0.0.1"
```
Comment (kirkonru) [inline docs/documentation/pages/DECKHOUSE-FAQ_RU.md:541]: --modules-filter="deckhouse-admin:v1.0.0`;`deckhouse-admin:v1.3.3`; `sds-drbd:v0.0.1" 

Do we need a space here?

**Review by kirkonru — APPROVED** [review]

File: docs/documentation/pages/DECKHOUSE-FAQ.md (hunk 5 above)
Comment (mvasl) [reply docs/documentation/pages/DECKHOUSE-FAQ.md:540]: This space is left there intentionally, to demonstrate that it is possible to use spaces between expressions

**Review by aleksey-su — APPROVED** [review]

**Pull request merged**